syntax="proto3";
package oceanim;

option cc_generic_services =true;

message UserAndIdNum {
    int64 user_id =1;
    int64 need_msgid_num =2;
}

message MsgIdRequest {
    repeated UserAndIdNum user_ids=1; // 第一个是sender
}

message MsgIds {
    int64 user_id =1;
    int64 start_msg_id = 2;
    int64 msg_id_num =3;
}

message MsgIdReply {
    repeated MsgIds msg_ids=1;
}

service IdGenService{
    // 获取ID， 支持批量获取， 多用户，多用户的的多ID
    rpc IdGenerate(MsgIdRequest) returns (MsgIdReply);
}